src/ppx/application.cpp

```diff
@@ -1133,6 +1149,36 @@ void Application::UpdateStandardSettings()
         mSettings.enableImGui = false;
         PPX_LOG_WARN("Headless or deterministic mode: disabling ImGui");
     }
+
+    if (mStandardOpts.pEnableFDM->GetValue()) {
```

Comment:
Yeath, I also tried making a string knob that could be set like `--shading-rate=fdm`, which seems more intuitive/easier to remember how to use, but dealing with the strings made the code a little more complicated.

It seems like the right answer is to have something like an enum type knob, where you could specify the valid strings and which value they mapped to.

I could switch to using strings for now, so that `--shading-rate=fdm` works, and then the internal implementation for that could be switched to an enum knob if that's ever added.

Reply:
I think that it might be possible to use a dropdown knob for this? Maybe something like the knob `pFullscreenQuadsType` and array of string&Enum `kFullscreenQuadsTypes` used in the graphics pipeline benchmark?

Reply:
...I just realized there might be a bug in `kFullscreenQuadsTypes` being all noise. But the idea is the same.

Reply:
You could define a knob for the shading rate and have it use `DropdownEntry<ShadingRateMode>`.

Side note: #387 (fix for the above)

Reply:
It looks like `KnobDropdown` is designed to allow the value to be changed through the UI, correct?

Shading rate mode cannot be changed after the application starts, so displaying a dropdown with different shading rate modes would be confusing.

With the current Knobs types, he best option I see is to use a `KnobFlag<string>`, with a validator to check that the string is one of the valid options ("" (default), "none", "fdm", "vrs"), and convert the string to `ShadingRateMode` here in `UpdateStandardSettings()`. Does that make sense?

Reply:
Yes, the current knob class structure is unfortunately confusing for the flag vs other classes division. 

I think `KnobFlag<string>` is a good option, and imo another option is to use `KnobDropdown` and just set the visibility to invisible so it's not shown in the UI. This would also help make the knob description message automatically list the options for you.

Reply:
Thanks, setting it to invisible would help. I went with a `KnobFlag<string>` because it still feels confusing to use a `KnobDropdown` for something that can't change and shouldn't be rendered as a dropdown. I also noticed that `KnobFlag` can display it's value without allowing the user modify it, which is a good fit here.